### DISCUSSION TITLE
Should we move to Zeitwerk for auto-loading?

### DISCUSSION THREAD

**Discussion by Drenmi** [rubocop/rubocop#11767] [Ideas]
Currently we have 2,271 require statements in Core alone.

For me, integrating Zeitwerk has been a great convenience in every project I've tried it, as it eliminates a whole class of human mistakes.

WDYT?

**Comment by dvandersluis** [comment]
I'm in favour, and it's something I've considered in the past as well. It might also allow us to speed up things in some situations where we currently load a ton of code that isn't used.

**Comment by koic** [comment]
Here's technical note. If autoloading is used, the `Registry` mechanism around `inherited` method needs to be taken care of.
